### PR TITLE
[KYUUBI #2203][FLINK] Support flink conf set by kyuubi conf file

### PR DESCRIPTION
**Pull request by deadwind4**
### _Why are the changes needed?_

Support setting Flink configuration by setting `kyuubi-default.conf`.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by pan3793** [comment]
Can we have a test for it?

**Comment by pan3793** [comment]
If I understand correctly, the behavior of engine-specific properties defined in `kyuubi-default.conf` is different in Spark and Flink. For Spark, we can define the properties as-is, e.g. define `spark.driver.memory` in `kyuubi-default.conf` has a same effect with defining `spark.driver.memory` in `spark-defaults.conf`, but for Flink, user need to add a prefix `flink` in `kyuubi-defaults.conf`, e.g. define `flink.jobmanager.memory.heap.size` in `kyuubi-default.conf` has a same effect with defining `jobmanager.memory.heap.size` in `flink-conf.yaml`. 

I'm fine with prefix design, but we need to document such behavior clearly otherwise users may confusing when they use both Spark and Flink engines.

**Comment by jiaoqingbo** [comment]
> 


> If I understand correctly, the behavior of engine-specific properties defined in `kyuubi-default.conf` is different in Spark and Flink. For Spark, we can define the properties as-is, e.g. define `spark.driver.memory` in `kyuubi-default.conf` has a same effect with defining `spark.driver.memory` in `spark-defaults.conf`, but for Flink, user need to add a prefix `flink` in `kyuubi-defaults.conf`, e.g. define `flink.jobmanager.memory.heap.size` in `kyuubi-default.conf` has a same effect with defining `jobmanager.memory.heap.size` in `flink-conf.yaml`.
> 
> I'm fine with prefix design, but we need to document such behavior clearly otherwise users may confusing when they use both Spark and Flink engines.

It's better to define a specific prefix for each engine and add it to the DOC, and take into account that we may add new engine support

**Comment by deadwind4** [comment]
> If I understand correctly, the behavior of engine-specific properties defined in `kyuubi-default.conf` is different in Spark and Flink. For Spark, we can define the properties as-is, e.g. define `spark.driver.memory` in `kyuubi-default.conf` has a same effect with defining `spark.driver.memory` in `spark-defaults.conf`, but for Flink, user need to add a prefix `flink` in `kyuubi-defaults.conf`, e.g. define `flink.jobmanager.memory.heap.size` in `kyuubi-default.conf` has a same effect with defining `jobmanager.memory.heap.size` in `flink-conf.yaml`.
> 
> I'm fine with prefix design, but we need to document such behavior clearly otherwise users may confusing when they use both Spark and Flink engines.

I will split a new issue to achieve Flink doc including this below.

**Comment by deadwind4** [comment]
> Can we have a test for it?

@pan3793 I have added an integration test case, thanks for your reminder.

File: integration-tests/kyuubi-flink-it/src/test/scala/org/apache/kyuubi/it/flink/FlinkSQLEngineSuite.scala [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  override val conf: KyuubiConf = KyuubiConf()
+    .set(ENGINE_TYPE, "FLINK_SQL")
+    .set(FRONTEND_THRIFT_BINARY_BIND_PORT, 10029)
+    .set("flink.parallelism.default", "6")
+
+  override protected def jdbcUrl: String = getJdbcUrl
+
+  test("set kyuubi conf into flink conf") {
+    withJdbcStatement() { statement =>
+      statement.execute("CREATE TABLE ods ( a_name STRING ) WITH ('connector' = 'datagen' )")
```
Comment (pan3793) [inline integration-tests/kyuubi-flink-it/src/test/scala/org/apache/kyuubi/it/flink/FlinkSQLEngineSuite.scala:35]: Does Flink support use `SET` command to echo the key value? If yes, we can simplify the test case

Comment (deadwind4) [reply integration-tests/kyuubi-flink-it/src/test/scala/org/apache/kyuubi/it/flink/FlinkSQLEngineSuite.scala]: You are right, I have used the `SET` statement to get echo conf.

**Review by pan3793 — APPROVED** [review]

**Review by yanghua — APPROVED** [review]
LGTM

**Comment by pan3793** [comment]
Thanks all, merging to master.

**Pull request closed without merging**